pkg/analyzer/analyzer.go

```diff
@@ -1,36 +1,76 @@
 package analyzer
 
 import (
+	"bytes"
+	"errors"
+	"flag"
 	"go/ast"
+	"go/printer"
+	"go/token"
+	"go/types"
+	"strings"
 
 	"golang.org/x/tools/go/analysis"
 	"golang.org/x/tools/go/analysis/passes/inspect"
 	"golang.org/x/tools/go/ast/inspector"
 )
 
+var (
+	ErrEmptyPattern = errors.New("pattern can't be empty")
```

Comment:
It is unclear which pattern cant be empty, you have to specify that error relates to exclusion patterns.